examples/ddpg/ddpg.py

```diff
@@ -43,137 +46,150 @@ def main(cfg: "DictConfig"):  # noqa: F821
-        # update weights of the inference policy
+
+        # Update weights of the inference policy
         collector.update_policy_weights_()
 
-        if r0 is None:
-            r0 = tensordict["next", "reward"].sum(-1).mean().item()
         pbar.update(tensordict.numel())
 
         tensordict = tensordict.reshape(-1)
         current_frames = tensordict.numel()
+        # Add to replay buffer
         replay_buffer.extend(tensordict.cpu())
         collected_frames += current_frames
 
-        # optimization steps
+        # Optimization steps
+        training_start = time.time()
         if collected_frames >= init_random_frames:
             (
                 actor_losses,
                 q_losses,
             ) = ([], [])
             for _ in range(num_updates):
-                # sample from replay buffer
+                # Sample from replay buffer
                 sampled_tensordict = replay_buffer.sample().clone()
 
+                # Compute loss
                 loss_td = loss_module(sampled_tensordict)
 
-                optimizer_critic.zero_grad()
-                optimizer_actor.zero_grad()
-
                 actor_loss = loss_td["loss_actor"]
                 q_loss = loss_td["loss_value"]
-                (actor_loss + q_loss).backward()
 
+                # Update critic
+                optimizer_critic.zero_grad()
+                q_loss.backward()
                 optimizer_critic.step()
-                q_losses.append(q_loss.item())
 
+                # Update actor
+                optimizer_actor.zero_grad()
+                actor_loss.backward()
                 optimizer_actor.step()
+
+                q_losses.append(q_loss.item())
                 actor_losses.append(actor_loss.item())
 
-                # update qnet_target params
+                # Update qnet_target params
                 target_net_updater.step()
 
-                # update priority
+                # Update priority
                 if prb:
                     replay_buffer.update_priority(sampled_tensordict)
 
-        rewards.append(
-            (i, tensordict["next", "reward"].sum().item() / env_per_collector)
-        )
-        train_log = {
-            "train_reward": rewards[-1][1],
-            "collected_frames": collected_frames,
-        }
-        if q_loss is not None:
-            train_log.update(
-                {
-                    "actor_loss": np.mean(actor_losses),
-                    "q_loss": np.mean(q_losses),
-                }
+        training_time = time.time() - training_start
+        episode_rewards = tensordict["next", "episode_reward"][
+            tensordict["next", "done"]
+        ]
+
+        # Logging
+        if len(episode_rewards) > 0:
+            episode_length = tensordict["next", "step_count"][
+                tensordict["next", "done"]
+            ]
+            logger.log_scalar(
+                "train/reward", episode_rewards.mean().item(), collected_frames
```

Comment:
ditto, let's put the logging in a separate function

Reply:
What do you think of the updated version for logging?